# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../libptp/gcc4.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur libptp2-1.1.0-orig/src/ptp-pack.c libptp2-1.1.0/src/ptp-pack.c
--- libptp2-1.1.0-orig/src/ptp-pack.c	2005-09-07 09:54:52.000000000 +0000
+++ libptp2-1.1.0/src/ptp-pack.c	2006-02-04 23:04:32.827336000 +0000
@@ -440,7 +440,7 @@
 		case PTP_DTC_STR:
 		{
 			uint8_t len;
-			(char *)(*value)=ptp_unpack_string(params,data,0,&len);
+			(*value)=(char *)ptp_unpack_string(params,data,0,&len);
 			break;
 		}
 	}
@@ -513,10 +513,10 @@
 		/* XXX: other int types are unimplemented */
 		/* XXX: int arrays are unimplemented also */
 		case PTP_DTC_STR:
-			(char *)dpd->FactoryDefaultValue = ptp_unpack_string
+			dpd->FactoryDefaultValue = (char *)ptp_unpack_string
 				(params,data,PTP_dpd_FactoryDefaultValue,&len);
 			totallen=len*2+1;
-			(char *)dpd->CurrentValue = ptp_unpack_string
+			dpd->CurrentValue = (char *)ptp_unpack_string
 				(params, data, PTP_dpd_FactoryDefaultValue + 
 				totallen, &len);
 			totallen+=len*2+1;
@@ -598,8 +598,8 @@
 			int i;
 			for(i=0;i<N;i++)
 			{
-				(char *)dpd->FORM.Enum.SupportedValue[i]=
-					ptp_unpack_string
+				dpd->FORM.Enum.SupportedValue[i]=
+					(char *)ptp_unpack_string
 					(params,data,PTP_dpd_FactoryDefaultValue
 					+totallen,&len);
 				totallen+=len*2+1;
